teuthology/task/kernel.py

```diff
@@ -906,12 +907,26 @@ def grub2_kernel_select_generic(remote, newversion, ostype):
     remote.run(args=['sudo', mkconfig, '-o', grubconfig, ])
     grub2conf = teuthology.get_file(remote, grubconfig, True)
     entry_num = 0
-    for line in grub2conf.split('\n'):
-        if line.startswith('menuentry'):
-            if newversion in line:
+    if '\nmenuitem ' not in grub2conf:
+        # okay, do the newer (el8) grub2 thing
+        grub2conf = remote.sh('sudo /bin/ls /boot/loader/entries || true')
```

Comment:
I'm not asking for a change, just wonder why there is a full path to `ls` required :-) I believe this is a legacy approach when there was legacy python function required a full path to executable module because it was running without shell :-)

Reply:
just paranoia on my part to make sure there isn't any colorization or ``/`` suffixes or whatever that you get with 'ls' aliases somethings.  those should't come up with a non-tty but just in case